app/models/user.rb

```diff
@@ -44,4 +44,23 @@ def can_follow?(user)
     user != self && !leaders.include?(user)
   end
 
+  def send_password_reset
+    generate_token(:password_reset_token)
+    self.password_reset_sent_at = Time.zone.now
+    save!
+    UserMailer.password_reset(self).deliver
+  end
+
+  def generate_token(column)
+    begin
+      self[column] = SecureRandom.urlsafe_base64
+    end until !User.exists?(column => self[column])
+  end
+
+  def cleanup_password_reset
+    self.password_reset_token = nil
+    self.password_reset_sent_at = nil
+    save!
```

Comment:
You could also use 1 line `update!(password_reset_token: nil, password_reset_sent_at: nil)`. Or if you would skip validations then `update_columns` should work as well.